.github/workflows/CI.yml

```diff
@@ -27,30 +27,62 @@ jobs:
         os:
           - ubuntu
           # Windows users, feel free to open a PR :)
+          # See issue #242, windows support would be welcome.
           # - windows
           - macos
-        exclude:
-          - os: windows # See issue #242, windows support would be welcome.
+        geos:
+          - '3.12.0'
+          - '3.11.2'
+          - '3.10.5'
+          - '3.9.4'
+          - '3.8.3'
     runs-on: ${{ matrix.os }}-latest
     continue-on-error: ${{ matrix.ruby == 'head' || matrix.os == 'windows' || matrix.os == 'macos' }}
-    name: Ruby ${{ matrix.ruby }} (${{ matrix.os }})
+    name: Ruby ${{ matrix.ruby }}, Geos ${{ matrix.geos }} (${{ matrix.os }})
     steps:
       - uses: actions/checkout@v3
-      - uses: ruby/setup-ruby@v1
-        with:
-          ruby-version: ${{ matrix.ruby }}
-          bundler-cache: true
-      - name: Install Geos (Linux)
+      - name: Install Packages (Linux)
         if: matrix.os == 'ubuntu'
         run: |
           sudo apt-add-repository ppa:ubuntugis/ubuntugis-unstable
-          sudo apt-get install -yqq libgeos-dev
-      - name: Install Geos (Mac)
+          sudo apt-get -yqq install curl libcurl4-openssl-dev libssl-dev cmake
+      - name: Install Packages (Mac)
         if: matrix.os == 'macos'
-        run: HOMEBREW_NO_INSTALL_CLEANUP=1 brew install geos
-      - name: Install Geos (Windows)
-        if: matrix.os == 'windows'
-        run: TODO
+        run: |
+          brew update
+          HOMEBREW_NO_INSTALL_CLEANUP=1 brew install --force cmake
+          brew upgrade cmake
+      - uses: actions/cache@v2
+        id: geos-cache
+        with:
+          path: ./geos-${{ matrix.geos }}
+          key: geos-${{ matrix.geos }}-${{ matrix.os }}
+      - name: Download and Compile Geos
+        if: steps.geos-cache.outputs.cache-hit != 'true'
+        run: |
+          wget -c https://download.osgeo.org/geos/geos-${{ matrix.geos }}.tar.bz2
+          tar -xf geos-${{ matrix.geos }}.tar.bz2
+          cd geos-${{ matrix.geos }}
+          mkdir build
+          cd build
+          cmake ..
+          cmake --build . -j2
+          cd ../../
+      - name: Install Geos
+        run: |
+          cd geos-${{ matrix.geos }}/build
+          sudo cmake --build . --target install
+          cd ../../
+      - name: ldconfig
+        if: matrix.os == 'ubuntu'
+        run: sudo ldconfig
+      - name: Set GEOS_LIBRARY_PATH (macos jruby)
+        if: matrix.os == 'macos' && matrix.ruby == 'jruby'
+        run: export GEOS_LIBRARY_PATH=/usr/local/lib/libgeos_c.dylib
+      - uses: ruby/setup-ruby@v1
```

Comment:
I tried on my mac, and I didn't need this `GEOS_LIBRARY_PATH` variable in my path

Reply:
yeah I'll try again I'm pretty stumped as to why this it isn't recognizing Geos. I shouldn't have to specify a path but clearly something's not configured correctly.